cmd/tether/serial/handshake.go

```diff
@@ -111,65 +125,68 @@ func readMultiple(conn net.Conn, b []byte) (int, error) {
 	return n, err
 }
 
-func HandshakeServer(conn net.Conn, interval time.Duration) {
+func HandshakeServer(ctx context.Context, conn net.Conn) error {
 	syn := make([]byte, 3)
 	synack := make([]byte, 3)
 	buf := make([]byte, 2)
 	ack := make([]byte, 2)
 
-	for {
-		// set the read deadline for timeout
-		// this has no effect on windows as the deadline is set at port open time
-		conn.SetReadDeadline(time.Now().Add(interval))
-
-		fmt.Println("reading syn")
-		// loop here until we get a valid syn opening. syn is 3 bytes as that will eventually
-		// syn us again if we're offset
-		if n, err := readMultiple(conn, syn); n != 2 || err != nil || syn[0] != SYN {
-			if err != nil {
-				fmt.Printf("Failed to read expected SYN: n=%d, err=%s\n", n, err)
-			} else if syn[0] != SYN {
-				fmt.Printf("Did not receive syn (read %v bytes): %#x != %#x\n", n, SYN, syn[0])
-				conn.Write([]byte{NAK})
-			} else {
-				fmt.Printf("Received syn but expected single sequence byte\n")
-			}
-
-			// to aid in debug we always dump the full handhsake
-			fmt.Printf("read %v bytes: ", n)
-			for i := 0; i < n; i++ {
-				fmt.Printf("%#x ", syn[i])
-			}
-			fmt.Println()
-
-			continue
-		}
-		fmt.Printf("Received syn: %#x\n", syn)
-
-		fmt.Println("writing synack")
-		synack[0] = ACK
-		synack[1] = syn[1] + 1
-		rand.Read(synack[2:])
-
-		conn.Write(synack)
+	// set the read deadline for timeout
```

Comment:
Yup - when time comes to revisit this code I may well have a top level context that gets used for shutting everything down. Need to see if that's an abuse of Cancel or will make things cleaner.